pkg/ocm/share/repository/nextcloud/nextcloud.go

```diff
@@ -400,7 +445,7 @@ func getUsername(user *userpb.User) string {
 		return user.Id.OpaqueId
 	}
 
-	return "empty-username"
+	return "nobody"
 }
 
 func (sm *Manager) do(ctx context.Context, a Action, username string) (int, []byte, error) {
```

Comment:
is the api in the nextcloud app always returning a json (when it sends a response body)?
Because we can accept an other parameter of type `any` and unmarshal from json if not nil.

```
func (sm *Manager) do(ctx context.Context, a Action, username string, r any) (int, error) {
    [...]
    if r != nil {
        if err := json.NewDecode(body).Decode(r); err != nil {
            handle error
        }
    }
}
```

Also we can do the same with the body in the request, as it seems that the caller is always encoding it to json and than call the `do` func.